core/src/autogluon/core/utils/utils.py

```diff
@@ -21,86 +21,95 @@
 from ..constants import BINARY, REGRESSION, MULTICLASS, SOFTCLASS, QUANTILE
 from ..metrics import accuracy, root_mean_squared_error, pinball_loss, Scorer
 
-logger = logging.getLogger(__name__)
-
 
+logger = logging.getLogger(__name__)
 
-# TODO: RespirceManager to replace these two hacks
-class _get_cpu_count:
-    def __init__(self):
-        self.mock = None
 
-    def __call__(self):
-        if self.mock is not None:
-            return self.mock
+class ResourceManager():
+    """Manager that fetches system related info"""
+    
+    @staticmethod
+    def get_cpu_count():
         return multiprocessing.cpu_count()
     
-
-class _get_gpu_count_all:
-    """
-    Attempts to get number of GPUs available for use via multiple means.
-    """
-    # FIXME: update to use only torch for TIMM or find a better GPU detection strategy
-    # FIXME: get_gpu_count by itself doesn't always work for Windows
-    def __init__(self):
-        self.mock = None
-
-    def __call__(self):
-        if self.mock is not None:
-            return self.mock
-        num_gpus = _get_gpu_count_cuda()
+    @staticmethod
+    def get_cpu_count_psutil(logical=True):
```

Comment:
we don't necessarily need to bind a different impl for the same function. If we see one is more trustworthy than the other, we should swap without impacting the caller

Reply:
@Innixma Do you remember why we have two ways retrieving num_cpus. Is it because psutil has more overhead but offer the logical option?

Reply:
Keep it for now, but this is a good question and we should figure out if one is strictly superior to the other

Reply:
Ideally if we can just use multiprocessing that would be best since it is native python